scripts/utils/termux_reuse_pr_build_artifacts.sh

```diff
@@ -0,0 +1,125 @@
+[ "$(sort -u <<< "$PRS" | wc -l)" -eq 1 ] || error_exit "push contains commits from more than one pull request, not performing CI fast path"
+
+PR_BASE_COMMIT="$(jq -r '.data.repository | to_entries[0].value.associatedPullRequests.nodes[0].baseRefOid' <<< "$RESPONSE")" || exit 0
+PR_HEAD_COMMIT="$(jq -r '.data.repository | to_entries[0].value.associatedPullRequests.nodes[0].headRefOid' <<< "$RESPONSE")" || exit 0
+[ -n "${PR_BASE_COMMIT:-}" ] || error_exit "failed to read associated PR base commit, not performing CI fast path"
+[ -n "${PR_HEAD_COMMIT:-}" ] || error_exit "failed to read associated PR head commit, not performing CI fast path"
+
+DIRS_REGEX="$(echo "$TERMUX_PACKAGE_DIRECTORIES" | paste -sd'|' -)" || exit 0
+
+# fetch PR commit tree
+git fetch origin $PR_BASE_COMMIT:ref/tmp/$PR_BASE_COMMIT || error_exit "failed to fetch PR base tree, not performing CI fast path"
+git fetch origin $PR_HEAD_COMMIT:ref/tmp/$PR_HEAD_COMMIT || error_exit "failed to fetch PR head tree, not performing CI fast path"
+
+# make sure nobody force-pushed after CI was invoked and before we fetched remote tree and requested GraphQL data
+# checking if changed files in PR are same as in this push should be enough.
+
+diff -q \
+    <(git diff $PR_BASE_COMMIT $PR_HEAD_COMMIT | sed -n -E '/^diff --git a\// { p; n; /^index /!p; b } ; p') \
+    <(git diff $OLD_COMMIT $HEAD_COMMIT | sed -n -E '/^diff --git a\// { p; n; /^index /!p; b } ; p') >/dev/null 2>&1 \
+|| error_exit "PR head does not match pushed commit changes, probably PR ref was force pushed right after PR was merged. Not performing CI fast path."
+
+
+# obtain merge-base
+PR_MERGE_BASE="$(git merge-base ref/tmp/$PR_BASE_COMMIT $HEAD_COMMIT)" || error_exit "failed to obtain PR merge base, not performing CI fast path"
+
+# obtain list of all files changed in master branch since this PR diverged
+PR_BASE_TO_HEAD_CHANGED_FILES="$(git diff-tree --name-only -r $PR_MERGE_BASE..$OLD_COMMIT)" || true
+
+# obtain list of all packages changed by this PR
+PR_CHANGED_PACKAGES="$(
+  git diff-tree --name-only -r "$OLD_COMMIT..$HEAD_COMMIT" \
+    | grep -E "^($DIRS_REGEX)/[^/]+/" \
+    | sed -E "s#^(($DIRS_REGEX)/[^/]+)/.*#\1#" \
+    | sort -u
+)" || true
+
+# obtain list of all buildsystem files changed since this PR diverged
+PR_BASE_TO_HEAD_CHANGED_BUILDSYSTEM_FILES="$(
+  grep -e "^scripts/" -e "^ndk-patches/" -e "^build-package.sh$" <<< "$PR_BASE_TO_HEAD_CHANGED_FILES"
+)" || true
+
+# obtain list of all packages changes since this PR diverged
+PR_BASE_TO_HEAD_CHANGED_PACKAGES="$(
+  echo "$PR_BASE_TO_HEAD_CHANGED_FILES" \
+    | grep -E "^($DIRS_REGEX)/[^/]+/" \
+    | sed -E "s#^(($DIRS_REGEX)/[^/]+)/.*#\1#" \
+    | sort -u
+)" || true
+
+# obtain list of all build dependencies of packages changed by this PR
+PR_CHANGED_PACKAGES_DEPS="$(
+  for i in ${PR_CHANGED_PACKAGES:-}; do
+    ./scripts/buildorder.py $i $TERMUX_PACKAGE_DIRECTORIES | awk '{print $NF}'
+  done | sort -u
+)" || true
```

Comment:
This is the last place I can mention it, so I'll put it here.
I stylistically prefer `|| :` over `|| true`.
This obviously doesn't change the semantics though.